aplib.net-demo/Assets/Scripts/Entities/Player/PlayerLogic.cs

```diff
@@ -6,11 +6,13 @@
 [RequireComponent(typeof(HealthComponent))]
 [RequireComponent(typeof(Movement))]
 [RequireComponent(typeof(ResetRigidbody))]
+[RequireComponent(typeof(EntitySound))]
 public class PlayerLogic : MonoBehaviour
 {
     private HealthComponent _healthComponent;
     private ResetRigidbody _resetRigidbody;
-    
+    private EntitySound _entitySound;
```

Comment:
We do not want to store the EntitySound and use it in this class, requiring it to enforce it existing on an enemy is fine, but it should be used within the weapons as they were, so that they can be used more easily and do not have to be used for every enemy type we may introduce

(Right now it's easier, melee weapon = melee sound, ranged weapon = ranged sound, check this out on the main branch!)

this comment applies to all uses of entity sound in this script

Reply:
(Copy paste but this applies to weapon+health component as well)